File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Btree/BTreeRemove.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Foundation;
using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Btree;

namespace Db4objects.Db4o.Internal.Btree
{
	/// <exclude></exclude>
	public class BTreeRemove : BTreeUpdate
	{
		public BTreeRemove(Transaction transaction, object obj) : base(transaction, obj)
		{
		}

		protected override void Committed(BTree btree)
		{
			btree.NotifyRemoveListener(new TransactionContext(_transaction, GetObject()));
		}

		public override string ToString()
		{
			return "(-) " + base.ToString();
		}

		public override bool IsRemove()
		{
			return true;
		}

		protected override object GetCommittedObject()
		{
			return No4.Instance;
		}

		protected override void AdjustSizeOnRemovalByOtherTransaction(BTree btree, BTreeNode
			 node)
		{
			// The size was reduced for this entry, let's change back.
			btree.SizeChanged(_transaction, node, +1);
		}

		protected override int SizeDiff()
		{
			return 0;
		}
	}
}

File: db4oj/core/src/com/db4o/internal/btree/BTreeRemove.java
/* Copyright (C) 2004   Versant Inc.   http://www.db4o.com */
package com.db4o.internal.btree;

import com.db4o.foundation.*;
import com.db4o.internal.*;

/**
 * @exclude
 */
public class BTreeRemove extends BTreeUpdate {
	
	public BTreeRemove(Transaction transaction, Object obj) {
        super(transaction, obj);
    }
    
    protected void committed(BTree btree){
        btree.notifyRemoveListener(new TransactionContext(_transaction, getObject()));
    }
    
    public String toString() {
        return "(-) " + super.toString();
    }
    
    public boolean isRemove() {
        return true;
    }

	protected Object getCommittedObject() {
		return No4.INSTANCE;
	}

    protected void adjustSizeOnRemovalByOtherTransaction(BTree btree, BTreeNode node) {
        // The size was reduced for this entry, let's change back.
        btree.sizeChanged(_transaction, node, +1);
    }
    
	protected int sizeDiff() {
		return 0;
	}
    
}
